### PR TITLE
use the configure_file() command to generate Path.h.  Fixes #1630

### LINKED ISSUE #1630: Reconfigure causes unnecessary rebuild
If I run make from drake/pod-build with all targets up to date, it runs very quickly, nothing rebuilds.  If I reconfigure cmake, then make will recompile some files and re-link almost everything, this takes a long time.  This often occurs when cmake configure rewrites generated files which are compile dependencies.  Sometimes this is easy to fix, sometimes you have to implement more robust dependency logic.  In this case, maybe it's from Path.h?  There might be others.  Path.h could be generated with the configure_file() command which properly tracks dependencies, rather than file(WRITE).

### PR DESCRIPTION
**Pull request by patmarion**
This avoids touching the file on successive reconfigures. The file will
only be touched if the cmake variables it depends on are modified.

Note, removes usage of ADDITIONAL_MAKE_CLEAN_FILES.

### CODE REVIEW

**Comment by patmarion** [comment]
Note @tkoolen this removes Path.h from ADDITIONAL_MAKE_CLEAN_FILES.  I'm not sure why you wanted make clean to remove Path.h from pod-build/generated/drake?  It's a configure time file, not a build time file.

File: drake/core/Path.h.in [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* This file is auto-generated by CMAKE.  Do not edit. */
+
+#include <string>
+
+namespace Drake {
+  const std::string& getDrakePath() {
+    static std::string root("@PROJECT_SOURCE_DIR@");
+    return root;
```
Comment (patmarion) [inline drake/core/Path.h.in]: I would have written this as

    return "/path";

Then let the compiler optimize it.  Is it documented that declaring a local static variable is more efficient?

Comment (patmarion) [inline drake/core/Path.h.in]: oh i just noticed it returns a const &.  ok.

**Comment by tkoolen** [comment]
Aha, good find! Yeah, the ADDITIONAL_MAKE_CLEAN_FILES thing is unnecessary. Also I believe this is still only a stopgap solution; baking the path into the executables doesn't play well with moving the whole drake install to a different folder (or different computer).

**Pull request merged**